docs/user-guide/config-best-practices.md

```diff
@@ -31,27 +31,131 @@ Documentation for other releases can be found at
-1. Use headless services for easy service discovery when you don't need kube-proxy load balancing. See [headless services](services.md#headless-services).
-1. Use kubectl delete rather than stop. Delete has a superset of the functionality of stop and stop is deprecated.
-1. If there is a viable alternative to naked pods (i.e. pods not bound to a controller), go with the alternative. Controllers are almost always preferable to creating pods (except for some `restartPolicy: Never` scenarios). A minimal Job is coming. See [#1624](http://issue.k8s.io/1624). Naked pods will not be rescheduled in the event of node failure.
-1. Put a version number or hash as a suffix to the name and in a label on a replication controller to facilitate rolling update, as we do for [--image](kubectl/kubectl_rolling-update.md). This is necessary because rolling-update actually creates a new controller as opposed to modifying the existing controller. This does not play well with version agnostic controller names.
-1. Put an object description in an annotation to allow better introspection.
+# Configuration Best Practices and Tips
+
+**Table of Contents**
+<!-- BEGIN MUNGE: GENERATED_TOC -->
+
+- [Configuration Best Practices and Tips](#configuration-best-practices-and-tips)
+  - [General Config Tips](#general-config-tips)
+  - ["Naked" Pods vs Replication Controllers and Jobs](#naked-pods-vs-replication-controllers-and-jobs)
+  - [Services](#services)
+  - [Using Labels](#using-labels)
+  - [Container Images](#container-images)
+  - [Using kubectl](#using-kubectl)
+
+<!-- END MUNGE: GENERATED_TOC -->
+
+
+This document is meant to highlight and consolidate in one place configuration best practices that are introduced throughout the user-guide and getting-started documentation and examples. This is a living document, so if you think of something that is not on this list but might be useful to others, please don't hesitate to file an issue or submit a PR.
+
+## General Config Tips
+
+- When defining configurations, specify the latest stable API version (currently v1).
+
+- Configuration files should be stored in version control before being pushed to the cluster. This allows a configuration to be quickly rolled back if needed, and will aid with cluster re-creation and restoration if necessary.
+
+- Write your configuration files using YAML rather than JSON. They can be used interchangeably in almost all scenarios, but YAML tends to be more user-friendly for config.
+
+- Group related objects together in a single file where this makes sense. This format is often easier to manage than separate files. See the [guestbook-all-in-one.yaml](../../examples/guestbook/all-in-one/guestbook-all-in-one.yaml) file as an example of this syntax.
+(Note that you can also call `kubectl create` on a directory of config files— see below for more detail).
```

Comment:
`kubectl create -f` isn't the only command that takes a directory of config files. Most of the kubectl commands do. Pointing out `kubectl create` may be misleading.